src/include/buffer.h

```diff
@@ -54,9 +56,12 @@ struct buffer_set_args
     struct buffer_set   *bsa_set;
     size_t               bsa_nbufs;
     size_t               bsa_buf_size;
+    void                *bsa_region;
+    size_t               bsa_region_size;
     void                *bsa_alt_source;
     size_t               bsa_alt_source_size;
-    size_t               bsa_alt_source_used;
+    uint8_t              bsa_use_alt_source_buf;
```

Comment:
I still don't see the material difference between `bsa_region` and `bsa_alt_source` but as it is, shouldn't these be mutually exclusive?  I do think it's better to simply use `bsa_alt_source` but I don't yet understand why that's not feasible.